### PR TITLE
Fuzzing: Add security fuzzer

### PR DESCRIPTION
**Pull request by AdamKorcz**
Adds a fuzzer that targets `istio.io/istio/security/pkg/pki/util.GenCSR`

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [x] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [x] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: tests/fuzz/security_fuzzer.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+func FuzzGenCSR(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	certOptions := util.CertOptions{}
+	err := f.GenerateStruct(&certOptions)
+	if err != nil {
+		return 0
+	}
+	_, _, _ = util.GenCSR(certOptions)
```
Comment (howardjohn) [inline tests/fuzz/security_fuzzer.go:30]: One concern: I think this is fairly expensive call (in CPU time) so we may end up not getting many iterations

**Review by howardjohn — COMMENTED** [review]
GenCSR is called on the client side rather than on the server side I believe, so it may not be as interesting as some of the other codepaths that run on the server side, which involve parsing arbitrary input (and are of course super security sensitive). Those may be a bit harder to fuzz possibly

File: tests/fuzz/security_fuzzer.go (hunk 1 above)
Comment (AdamKorcz) [reply tests/fuzz/security_fuzzer.go:45]: Locally it runs alright. I will keep an eye out for the stats from runs by OSS-fuzz.

**Comment by AdamKorcz** [comment]
> GenCSR is called on the client side rather than on the server side I believe, so it may not be as interesting as some of the other codepaths that run on the server side, which involve parsing arbitrary input (and are of course super security sensitive). Those may be a bit harder to fuzz possibly

I have included a fuzzer for server side targets.

**Comment by AdamKorcz** [comment]
Checking in to see if this one can be considered ready to be merged.

**Comment by AdamKorcz** [comment]
@howardjohn @jacob-delgado Kindly requesting review of this to see if it can be approved.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**